### PR TITLE
[freexl] Fix mingw, cleanup

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  This PR fixes the mingw, modernizes the portfile, and adds the libiconv dependency to the pc file.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

**Comment by dg0yt** [comment]
CI is rebuilding a lot of packages now, just after update of `master`. Let's hope we don't get too many unrelated regressions.

**Comment by dg0yt** [comment]
rtaudio:x64-osx CI failure: unrelated.

File: ports/freexl/portfile.cmake [hunk 1]
```diff
@@ -37,55 +38,33 @@ if (VCPKG_TARGET_IS_WINDOWS)
...
+        SOURCE_PATH "${SOURCE_PATH}"
+        AUTOCONFIG
     )
-    
     vcpkg_install_make()
-    vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES m)
-    
-    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
-    file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/freexl RENAME copyright)
+    vcpkg_fixup_pkgconfig()
```
Comment (JackBoosY) [inline ports/freexl/portfile.cmake:65]: Does this not need `SYSTEM_LIBRARIES m` now?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/freexl/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/freexl/portfile.cmake:65]: No port does: 
https://github.com/microsoft/vcpkg/blob/7bdf2f8e853a476f96f3d1ba3be409b65822a29d/scripts/cmake/vcpkg_fixup_pkgconfig.cmake#L34-L35

File: ports/freexl/portfile.cmake [hunk 2]
```diff
@@ -37,55 +38,33 @@ if (VCPKG_TARGET_IS_WINDOWS)
...
 
-else()# Other build system
-    message(FATAL_ERROR "Unsupported build system.")
 endif()
 
-message(STATUS "Packaging ${TARGET_TRIPLET} done")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/freexl" RENAME copyright)
```
Comment (JackBoosY) [inline ports/freexl/portfile.cmake:70]:
```suggestion
file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/freexl/portfile.cmake (hunk 2 above)
Comment (dg0yt) [reply ports/freexl/portfile.cmake]: Done.

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Yep, this LGTM. Thanks for the awesome cleanup!

**Pull request merged**